droidlet/interpreter/robot/tasks.py

```diff
@@ -334,6 +335,31 @@ def step(self):
                         agent.memory.untag(mmid, "_in_inventory")
 
 
+class TrajectorySaverTask(Task):
+    def __init__(self, agent, task_data):
```

Comment:
at a high level, why should this be a Task?

Reply:
I wanted the ability to save trajectories to be a feature of the Examine/Explore tasks. This way, the inherited tasks simply call `super().step()` to save stuff. The alternative would have a bit more verbose too (ie define a TrajectoryDataSaver object inside each Examine/Explore task and calling save) and I wanted to avoid that too.